src/test/java/cloud/prefab/client/integration/DynamicTestBuilder.java

```diff
@@ -0,0 +1,76 @@
+package cloud.prefab.client.integration;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import cloud.prefab.client.Options;
+import cloud.prefab.client.PrefabCloudClient;
+import cloud.prefab.client.integration.Expectation;
+import cloud.prefab.client.integration.Scenario;
+import cloud.prefab.domain.Prefab;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.function.Executable;
+
+public class DynamicTestBuilder {
+
+  public static DynamicTest build(
+    Map<String, Object> testData,
+    String version,
+    String function
+  ) {
+    Object input = testData.get("input");
```

Comment:
I kinda want this to be:

`Scenario scenario = new Scenario(testData.get("input");
Expectation expectation = new Expectation(testData.get("expected"))`

Reply:
This will require some refactoring in the other clients and the syntax, but I was able to get closer to this for now.